### PR TITLE
Fixed build with -DQT_NO_CAST_FROM_ASCII

### LINKED ISSUE #170: Build fails with -DQT_NO_CAST_FROM_ASCII
When `-DQT_NO_CAST_FROM_ASCII` is defined, you cant call `QChar(' ')`(char) and `QString("String")`(const char*).
To avoid the QChar error we could use `QChar(u' ')` but this requires Qt >= 5.10.  
For QString we could use `QStringLiteral`.  

I created a issue because i don't want to create a pull request that requires Qt 5.10 without having your OK.

### PR DESCRIPTION
**Pull request by software-made-easy**
Closes #170

### CODE REVIEW

File: markdownhighlighter.cpp [hunk 1]
```diff
@@ -241,7 +240,7 @@ void MarkdownHighlighter::initHighlightingRules() {
         HighlightingRule rule(HighlighterState::TrailingSpace);
         rule.pattern = QRegularExpression(QStringLiteral("( +)$"));
         // waqar144: don't use QStringLiteral here.
-        rule.shouldContain = QString(" \0");
+        rule.shouldContain = QString::fromLatin1(" \0");
```
Comment (software-made-easy) [inline markdownhighlighter.cpp:243]: I'm not sure if this works

Comment (Waqar144) [reply markdownhighlighter.cpp:243]: It will work.

**Comment by pbek** [comment]
Unfortunately this PR breaks all openSUSE_Leap builds, like https://build.opensuse.org/package/binaries/home:pbek:QOwnNotes/desktop/openSUSE_Leap_42.3_Ports and Debian 9 builds, like https://build.opensuse.org/package/live_build_log/home:pbek:QOwnNotes/desktop/Debian_9.0/x86_64.

**Comment by pbek** [comment]
Fedora broken too... As is the old Windows XP build... https://ci.appveyor.com/project/pbek/qownnotes/builds/44736536

**Comment by Waqar144** [comment]
The error was trivial, could have been fixed instead of full revert, would have been less effort too ¯\_(ツ)_/¯.

**Comment by pbek** [comment]
Getting all build systems to build a release again without breaking a lot of them again is ALWAYS a lot of work regardless of what code changes need to take place upfront. Reverting was a trivial amount of work.

**Comment by pbek** [comment]
> Getting all build systems to build a release again without breaking a lot of them again

I working on nothing else since almost an hour...

**Comment by Waqar144** [comment]
> Getting all build systems to build a release again without breaking a lot of them again is ALWAYS a lot of work

Yeah, but nothing special and nothing that hasn't been done before. On the other hand we lost a feature.

**Comment by Waqar144** [comment]
Also, we can do Qt5.6 ci testing I think. Binaries are up here: https://download.qt.io/new_archive/qt/5.6/5.6.3/

One just needs to download and extract it and then push it to a github repo. Then we can just pull it during CI run and use this Qt to build QOwnNotes.

In the worst case, I can do a build before release to ensure it compiles.

**Comment by pbek** [comment]
I'd be very happy if someone could integrate the old Qt archive into the CI build matrix! I'd even attempt to integrate that into the QOwnNotes repo once it runs here (will take more work and time tho)...

And If the last PR is easy to fix we can happily accept a new PR once there is a way to test it. 😅

**Pull request merged**